src/journal/journal-file.c

```diff
@@ -3312,7 +3314,11 @@ int journal_file_open(
         }
 
         if (f->fd < 0) {
-                f->fd = open(f->path, f->flags|O_CLOEXEC, f->mode);
+                /* We pass O_NONBLOCK here, so that in case somebody pointed us to some block device or so, we likely
+                 * fail quickly than block for long. For regular files O_NONBLOCK has no effect, hence it doesn't hurt
+                 * in that case. */
+
+                f->fd = open(f->path, f->flags|O_CLOEXEC|O_NONBLOCK, f->mode);
```

Comment:
Hmm, open(2) says:
> Note that this flag has no effect for regular files  and  block  devices

so the comment seems wrong, but what's worse it also says
> Since  O_NONBLOCK semantics might eventually be implemented, applications should not depend upon blocking behavior when specifying this flag for regular files and block devices.

which means that (potentially), our code might stop working in the future. Since this is in library code, that could even be potentially running on non-linux systems, we shouldn't depend on the implementation details of the kernel.

Reply:
I figure i should say char device, not block device... any maybe also list fifos or so...

well, the stuff is pretty linux specific anyway. if somebody indeed ports that to other OSes, then they need to investigate this kind of stuff anyway... I am pretty sure it's highly unlikely O_NONBLOCK will ever be implemented for regular files on Linux, given that usage like the above is seen quite often.

The primary reason I added this is that we iterate trhough a directory tree and open all potentially useful journal files we see there. i.e. it's not that the knowledgeable admin would never point us to weird device files and such, it's that we go out on our own to find stuff to open and open it, and that makes things dangerous, and something we should protect us from...

Reply:
OK.